File: webapp/js/maptools.js
function getNormalizedCoord(coord, zoom) {
    var y = coord.y;
    var x = coord.x;

    // tile range in one direction range is dependent on zoom level
    // 0 = 1 tile, 1 = 2 tiles, 2 = 4 tiles, 3 = 8 tiles, etc
    var tileRange = 1 << zoom;

    /**
    * If we don't want to repeat in any direction
    * simply return null.
    */
    // repeat across y-axis (vertically)
    if (y < 0 || y >= tileRange) return null
    // repeat across x-axis (horizontally)
    if (x < 0 || x >= tileRange) {
      x = (x % tileRange + tileRange) % tileRange
    }

    return {
      x: x,
      y: y
    };
}

function bound(value, opt_min, opt_max) {
  if (opt_min != null) value = Math.max(value, opt_min);
  if (opt_max != null) value = Math.min(value, opt_max);
  return value;
}

function degreesToRadians(deg) {
  return deg * (Math.PI / 180);
}

function radiansToDegrees(rad) {
  return rad / (Math.PI / 180);
}

/** @constructor */
function MercatorProjection() {
  this.pixelOrigin_ = new google.maps.Point(TILE_SIZE / 2,
      TILE_SIZE / 2);
  this.pixelsPerLonDegree_ = TILE_SIZE / 360;
  this.pixelsPerLonRadian_ = TILE_SIZE / (2 * Math.PI);
}

MercatorProjection.prototype.fromLatLngToPoint = function(latLng, opt_point) {
  var me = this;
  var point = opt_point || new google.maps.Point(0, 0);
  var origin = me.pixelOrigin_;
  point.x = origin.x + latLng.lng() * me.pixelsPerLonDegree_;
  // Truncating to 0.9999 effectively limits latitude to 89.189. This is
  // about a third of a tile past the edge of the world tile.
  var siny = bound(Math.sin(degreesToRadians(latLng.lat())), -0.9999,
      0.9999);
  point.y = origin.y + 0.5 * Math.log((1 + siny) / (1 - siny)) * -me.pixelsPerLonRadian_;

  return point;
};

MercatorProjection.prototype.fromPointToLatLng = function(point) {
  var me = this;
  var origin = me.pixelOrigin_;
  var lng = (point.x - origin.x) / me.pixelsPerLonDegree_;
  var latRadians = (point.y - origin.y) / -me.pixelsPerLonRadian_;
  var lat = radiansToDegrees(2 * Math.atan(Math.exp(latRadians)) -
      Math.PI / 2);
  return new google.maps.LatLng(lat, lng);
};

File: app/SFUMaps/app/src/main/java/me/gurinderhans/sfumaps/MapTools.java
package me.gurinderhans.sfumaps;

import android.graphics.PointF;

import com.google.android.gms.maps.model.LatLng;

/**
 * Created by ghans on 2/9/15.
 */
public class MapTools {

    static PointF pixelOrigin_ = new PointF(AppConstants.TILE_SIZE / 2, AppConstants.TILE_SIZE / 2);
    static double pixelsPerLonDegree_ = AppConstants.TILE_SIZE / 360;
    static double pixelsPerLonRadian_ = AppConstants.TILE_SIZE / (2 * Math.PI);

    public static PointF fromLatLngToPoint(LatLng latLng) {
        PointF point = new PointF(0, 0);
        PointF origin = pixelOrigin_;

        point.x = (float) (origin.x + latLng.longitude * pixelsPerLonDegree_);

        // Truncating to 0.9999 effectively limits latitude to 89.189. This is
        // about a third of a tile past the edge of the world tile.
        double siny = bound(Math.sin(degreesToRadians(latLng.latitude)), -0.9999, 0.9999);

        point.y = (float) (origin.y + 0.5 * Math.log((1 + siny) / (1 - siny)) * -pixelsPerLonRadian_);
        return point;
    }

    public static LatLng fromPointToLatLng(PointF point) {
        PointF origin = pixelOrigin_;
        double lng = (point.x - origin.x) / pixelsPerLonDegree_;
        double latRadians = (point.y - origin.y) / -pixelsPerLonRadian_;
        double lat = radiansToDegrees(2 * Math.atan(Math.exp(latRadians)) - Math.PI / 2);
        return new LatLng(lat, lng);
    }

    public static double bound(double value, double opt_min, double opt_max) {
        if (opt_min != 0) return Math.max(value, opt_min);
        if (opt_max != 0) return Math.min(value, opt_max);
        return -1;
    }

    private static double degreesToRadians(double deg) {
        return deg * (Math.PI / 180);
    }

    private static double radiansToDegrees(double rad) {
        return rad / (Math.PI / 180);
    }
}